lib/onegini_event_listener.dart

```diff
@@ -1,160 +1,101 @@
+  void n2fEventInitCustomRegistration(
+      OWCustomInfo? customInfo, String providerId) {
+    broadcastEvent(InitCustomRegistrationEvent(customInfo, providerId));
   }
 
   @override
   void n2fEventFinishCustomRegistration(
       OWCustomInfo? customInfo, String providerId) {
-    eventFinishCustomRegistration(_context, customInfo, providerId);
+    broadcastEvent(FinishCustomRegistrationEvent(customInfo, providerId));
   }
 
+  /// Pin Authentication related events
   @override
-  void n2fEventInitCustomRegistration(
-      OWCustomInfo? customInfo, String providerId) {
-    eventInitCustomRegistration(_context, customInfo, providerId);
+  void n2fOpenPinAuthentication() {
+    broadcastEvent(OpenPinAuthenticationEvent());
   }
 
   @override
-  void n2fHandleRegisteredUrl(String url) {
-    print("hello from handleRegisteredUrl");
-    handleRegisteredUrl(_context, url);
+  void n2fClosePinAuthentication() {
+    broadcastEvent(ClosePinAuthenticationEvent());
   }
 
   @override
-  void n2fNextAuthenticationAttempt(
+  void n2fNextPinAuthenticationAttempt(
       OWAuthenticationAttempt authenticationAttempt) {
-    nextAuthenticationAttempt(
-        _context,
-        AuthenticationAttempt(
-            failedAttempts: authenticationAttempt.failedAttempts,
-            maxAttempts: authenticationAttempt.maxAttempts,
-            remainingAttempts: authenticationAttempt.remainingAttempts));
+    broadcastEvent(NextPinAuthenticationAttemptEvent(authenticationAttempt));
   }
 
+  /// Fingerprint related events
   @override
-  void n2fOpenAuthOtp(String? message) {
-    openAuthOtp(_context, message != null ? message : "");
+  void n2fShowScanningFingerprint() {
+    broadcastEvent(ShowScanningFingerprintEvent());
   }
 
   @override
   void n2fOpenFingerprintScreen() {
-    openFingerprintScreen(_context);
+    broadcastEvent(OpenFingerprintEvent());
   }
 
   @override
-  void n2fOpenPinAuthenticator() {
-    openPinAuthenticator(_context);
-  }
-
-  @override
-  void n2fOpenPinRequestScreen() {
-    openPinRequestScreen(_context);
+  void n2fCloseFingerprintScreen() {
+    broadcastEvent(CloseFingerprintEvent());
   }
 
   @override
-  void n2fOpenPinScreenAuth() {
-    openPinScreenAuth(_context);
+  void n2fNextFingerprintAuthenticationAttempt() {
+    broadcastEvent(NextFingerprintAuthenticationAttempt());
   }
 
+  /// OTP Mobile authentication related events
   @override
-  void n2fReceivedFingerprint() {
-    receivedFingerprint(_context);
+  void n2fOpenAuthOtp(String? message) {
+    broadcastEvent(OpenAuthOtpEvent(message ?? ""));
   }
 
   @override
-  void n2fShowScanningFingerprint() {
-    showScanningFingerprint(_context);
+  void n2fCloseAuthOtp() {
+    broadcastEvent(CloseAuthOtpEvent());
   }
 
-  @override
-  void n2fEventPinNotAllowed(OWOneginiError error) {
-    pinNotAllowed(error);
+  /// Helper method
+  void broadcastEvent(OWEvent event) {
+    Onegini.instance.userClient.owEventStreamController.sink.add(event);
```

Comment:
Can we make this method private and also add the streamController through the constructor?

Reply:
done